### PR TITLE
[CPU] Renamed CPU plugin to ov_intel_cpu_plugin

### PR DESCRIPTION
**Pull request by vladislav-volkov**
### Details:
 - CPU plugin renamed to ov_intel_cpu_plugin and moved to new place

### Tickets:
 - CVS-71045

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
what's about moving on mkldnn to a plugin folder?

**Comment by vladislav-volkov** [comment]
> what's about moving on mkldnn to a plugin folder?
ok, I'll move it to internal directory

File: scripts/CMakeLists.txt [hunk 1]
```diff
@@ -10,15 +10,19 @@ ie_shellcheck_process(DIRECTORY "${OpenVINO_SOURCE_DIR}"
                       SKIP "${OpenVINO_SOURCE_DIR}/bin"
                            "${OpenVINO_SOURCE_DIR}/build"
                            "${OpenVINO_SOURCE_DIR}/thirdparty"
+                           "${OpenVINO_SOURCE_DIR}/src/plugins/intel_cpu/thirdparty"
```
Comment (dmitry-gorokhov) [inline scripts/CMakeLists.txt:13]: Why do we need changes in this file?

File: tools/pot/.gitignore [hunk 2]
```diff
@@ -105,3 +105,6 @@ ENV/
 
 # snapshots
 *.tar
+
+# Autogenerated version file
```
Comment (dmitry-gorokhov) [inline tools/pot/.gitignore]: I think it is a valuable change, but it shouldn't be part of this PR.

Comment (vladislav-volkov) [reply tools/pot/.gitignore]: removed

File: scripts/CMakeLists.txt (hunk 1 above)
Comment (vladislav-volkov) [reply scripts/CMakeLists.txt:13]: I removed all changes except /src/plugins/intel_cpu/thirdparty, but the locally spellchecker is failed on
"${IE_MAIN_SOURCE_DIR}/temp"
"${OpenVINO_SOURCE_DIR}/scripts/install_dependencies/install_NCS_udev_rules.sh"
"${OpenVINO_SOURCE_DIR}/src/tests/functional/plugin/myriad/run_tests_myriad_multistick.sh".
I think, it happened due to my spellchecker version.

> shellcheck --version
ShellCheck - shell script analysis tool
version: 0.7.0

**Review by dmitry-gorokhov — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by vladislav-volkov** [comment]
Tests on product-configs repository: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/732

**Comment by azhogov** [comment]
CI passed: https://github.com/intel-innersource/frameworks.ai.openvino.ci.product-configs/pull/732
